exercises/circular-buffer/src/example/java/BufferIOException.java

```diff
@@ -0,0 +1,5 @@
+public class BufferIOException extends RuntimeException {
```

Comment:
This should extend `Exception` instead of `RuntimeException`. `RuntimeException` should be for errors  you can't fix, generally programmer errors where the best thing to do is for the program to crash. If you want the exception to be handled then extend `Exception` because the compiler will force you to handle the exception. See [this article](http://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/) for some more information about it :)